.github/workflows/rust-mirai.yml

```diff
@@ -0,0 +1,64 @@
+name: Rust
+
+on:
+  pull_request:
+    types: ['opened', 'edited', 'reopened', 'synchronize']
+
+jobs:
+  install_and_run_MIRAI:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install Z3
+      uses: pavpanchekha/setup-z3@v1.2
+    - name: Checkout MIRAI
+      uses: actions/checkout@v2
+      with:
+        repository: 'facebookexperimental/MIRAI'
+        path: 'MIRAI'
+    - name: Install MIRAI
+      working-directory: ./MIRAI
+      run: |
+        rustup component add rustc-dev
+        RUSTFLAGS='-Clink-arg=-L./binaries -Clink-arg=-lstdc++' cargo install --path ./checker
+    - name: Checkout libra
+      uses: actions/checkout@v2
+      with:
+        path: 'libra'
+    - name: Build libra master then checkout PR
+      working-directory: ./libra
+      run: |
+        rustup override set `cat ../MIRAI/rust-toolchain`
+        rustup component add rustc-dev
+        git fetch origin master
+        git checkout master
```

Comment:
Can we leave a note here why we need to build master first?

Reply:
Also, it may be the case that the commit cannot be automatically rebased on current master. It would be better to check out the branch's parent commit instead. Or at least check if the commit still rebases cleanly.